test/distributed/fsdp/test_wrap.py

```diff
@@ -133,14 +370,20 @@ def test_auto_wrap_smoke_test(self):
         os.environ["MASTER_PORT"] = str(find_free_port())
         torch.distributed.init_process_group(backend="nccl", rank=0, world_size=1)
 
+        # NOTE: We move model to CUDA after init with FSDP to simulate real use
+        # cases where full model cannot be loaded onto GPU, but their shards can.
         try:
-            with enable_wrap(wrapper_cls=FSDP):
-                sequential = nn.Sequential(
-                    nn.Linear(5, 5), nn.Linear(5, 5), nn.Sequential(nn.Linear(5, 5), nn.Linear(5, 5))
-                )
-                my_auto_wrap_policy = functools.partial(default_auto_wrap_policy, min_num_params=40)
-                model = auto_wrap(sequential, auto_wrap_policy=my_auto_wrap_policy)
-            model.to(device)
+            sequential = TestFSDPWrap.NestedSequentialModel.get_model(cuda=False)
```

Comment:
probably useful to test with CUDA before and CUDA after as well, though afaik CUDA after is a bit more important, as the intial model might fit on CPU but not GPU.

Reply:
sounds good